client/indexer-db/src/models/bsp.rs

```diff
@@ -90,4 +92,47 @@ impl Bsp {
             .await?;
         Ok(())
     }
+
+    pub async fn get_by_onchain_bsp_id<'a>(
+        conn: &mut DbConnection<'a>,
+        onchain_bsp_id: String,
+    ) -> Result<Self, diesel::result::Error> {
+        let bsp = bsp::table
+            .filter(bsp::onchain_bsp_id.eq(onchain_bsp_id))
+            .first(conn)
+            .await?;
+        Ok(bsp)
+    }
+
+    pub async fn get_by_account<'a>(
+        conn: &mut DbConnection<'a>,
+        account: String,
+    ) -> Result<Self, diesel::result::Error> {
+        let bsp = bsp::table
+            .filter(bsp::account.eq(account))
+            .first(conn)
+            .await?;
+        Ok(bsp)
+    }
+}
+
+#[derive(Debug, Queryable, Insertable, Selectable)]
+#[diesel(table_name = bsp_file)]
+pub struct BspFile {
+    pub bsp_id: i32,
+    pub file_id: i32,
+}
+
+impl BspFile {
```

Comment:
From what I understand, this links a `File` to a `bsp_id`, so that we only store the `File` information once, but then we have several BSPs that reference it as being a part of their forests, right?

Reply:
Yes. 
We will have multiple entries here, 1 for each BSP storing the File. Also there are delete triggers setup in Postgres so that if any of the BSP or the File is deleted the relationship entry from this table is also deleted.